movai_core_shared/core/zmq/zmq_manager.py

```diff
@@ -0,0 +1,85 @@
+from logging import getLogger
+
+from movai_core_shared.core.zmq.zmq_base import ZMQBase
+from movai_core_shared.core.zmq.zmq_client import ZMQClient, AsyncZMQClient
+from movai_core_shared.core.zmq.zmq_subscriber import ZMQSubscriber, AsyncZMQSubscriber
+from movai_core_shared.core.zmq.zmq_publisher import ZMQPublisher, AsyncZMQPublisher
+from movai_core_shared.core.zmq.zmq_helpers import generate_zmq_identity
+from movai_core_shared.exceptions import ArgumentError
+
+
+class ZMQManager:
+    _logger = getLogger("ZMQManager")
+    _clients = {}
+    _async_clients = {}
+    _publishers = {}
+    _async_publishers = {}
+    _subscribers = {}
+    _async_subscribers = {}
+
+    zmq_types = {
+        ZMQClient.__name__: "dealer",
+        AsyncZMQClient.__name__: "dealer",
+        ZMQPublisher.__name__: "pub",
+        AsyncZMQPublisher.__name__: "pub",
+        ZMQSubscriber.__name__: "sub",
+        AsyncZMQSubscriber.__name__: "sub"
+    }
+    
+    @classmethod
+    def validate_server_addr(cls, server_addr: str):
+        if not isinstance(server_addr, str):
+            error_msg = "The argument server_addr must be of type string"
+            cls._logger.error(error_msg)
+            raise ArgumentError(error_msg)
+
+    def validate_zmq_type(cls, zmq_object: ZMQBase):
```

Comment:
why do we need this function ? 
I don't find it's use case in your code

Reply:
got removed